File: app/src/main/java/me/zipi/navitotesla/db/ConditionDao.java
package me.zipi.navitotesla.db;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface ConditionDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertCondition(ConditionEntity entity);

    @Query("SELECT * FROM condition WHERE type = :type")
    LiveData<List<ConditionEntity>> findCondition(String type);

    @Query("SELECT * FROM condition WHERE type = :type")
    List<ConditionEntity> findConditionSync(String type);

    @Query("SELECT * FROM condition WHERE type = :type AND name = :name")
    ConditionEntity findConditionByNameSync(String type, String name);

    @Delete
    void delete(ConditionEntity entity);

}

File: app/src/main/kotlin/me/zipi/navitotesla/db/ConditionDao.kt
package me.zipi.navitotesla.db

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface ConditionDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertCondition(entity: ConditionEntity)


    @Query("SELECT * FROM condition WHERE type = :type")
    suspend fun findCondition(type: String): List<ConditionEntity>

    @Query("SELECT * FROM condition WHERE type = :type AND name = :name")
    suspend fun findConditionByName(type: String, name: String): ConditionEntity?

    @Delete
    suspend fun delete(entity: ConditionEntity)
}